ethsigner/src/main/java/tech/pegasys/ethsigner/CommandLineConfig.java

```diff
@@ -100,14 +100,20 @@
       arity = "1")
   private final Integer httpListenPort = 8545;
 
-  @SuppressWarnings("FieldMayBeFinal") // Because PicoCLI requires Strings to not be final.
+  @SuppressWarnings("FieldMayBeFinal")
   @Option(
       names = {"--chain-id"},
       description = "The Chain Id that will be the intended recipient for signed transactions",
       required = true,
       arity = "1")
   private long chainId;
 
+  @Option(
```

Comment:
ethsigner needing a data-dir is really disappointing given its testing-specific necessity.

Reply:
Agreed, but I'm all ears for a better way to achieve state sharing between the framework and EthSigner.

It's a problem with not knowing the ports, as it rules out tcp communication (until the ports are known) ...and it is a feature flag option.